### PR TITLE
TS-4771: Remove config, stats, and docs for ICP

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/643/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/539/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/644/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/540/ for details.

File: mgmt/RecordsConfig.cc [hunk 1]
```diff
@@ -1299,6 +1299,9 @@ static const RecordElement RecordsConfig[] =
   ,
   {RECT_CONFIG, "proxy.config.ssl.wire_trace_server_name", RECD_STRING, NULL , RECU_DYNAMIC, RR_NULL, RECC_STR, ".*", RECA_NULL}
   ,
+  {RECT_CONFIG, "proxy.config.ssl.cert.load_elevated", RECD_INT, "0", RECU_RESTART_TS, RR_NULL, RECC_INT, "[0-1]", RECA_READ_ONLY}
+  ,
+
```
Comment (zwoop) [inline mgmt/RecordsConfig.cc]: Should this really be in this patch set ?

Comment (zwoop) [inline mgmt/RecordsConfig.cc]: Oh, never mind, it got moved up.

Comment (bryancall) [inline mgmt/RecordsConfig.cc]: Yeah, it looked out of place.

**Comment by zwoop** [comment]
Couple of minor question / issues:

1) There are some references to the ICP Via headers in the cmd/traffic_via/tests directory. They should go away maybe? Or do we still generate this ?

2) This doesn't remove the actual code as far as I can tell, is there another PR for that?

3) I found one line still using a (now missing) ICP define:

example/redirect-1/redirect-1.c:    else if (0 == strncmp(txn_method, TS_HTTP_METHOD_ICP_QUERY, length))

**Comment by bryancall** [comment]
This pull request disables ICP, removes the configuration, APIs, and stats.  I am breaking up removing ICP into pieces. Being that it is late in the release cycle it might be best to rip out the ICP code for the 7.1.0 release.

**Comment by zwoop** [comment]
:+1:

**Pull request merged**